revised test runner and generator

sevko opened the pull request:
Wrote a custom test-runner and output generator which prints less noisy output to the terminal and will allow us to, in the future, easily re-format output (into, say, an email) and gather statistics (like the number of passing/failing tests).

Closes #8.

dianashk commented on test.js:
```diff
@@ -0,0 +1,204 @@
+ * URLs for the various Pelias APIs out in the wild. Can be specified as a
+ * command-line argument (see `runTests()`).
+ */
+var PELIAS_ENDPOINTS = {
```
these should live in a config file somewhere, and you should be able to specify a different url as well.

sevko replied:
Could be `pelias-config`?

dianashk replied:
pelias-config is good

dianashk commented on test.js:
```diff
@@ -0,0 +1,204 @@
+  }
+  else {
+    console.error(
+      apiUrl, 'is not a recognized endpoint. Try:',
```
apiUrl won't have a value in the else clause, since it gets assigned in the if.

sevko replied:
Oh, good catch. Was meant to be `commander.endpoint`.

dianashk commented on test.js:
```diff
@@ -0,0 +1,200 @@
+    results: []
+  };
+
+  var startTime = new Date().getTime();
```
should probably use microtime for this kind of stuff

sevko replied:
It's not really a benchmark and added as somewhat of stat-sugar, so I'd opt for the simplicity of `new Date()`. Plus, it's not totally accurate because these tests get executed in parallel (meaning `startTime` gets initialized immediately, while the API processes results more or less sequentially and the time per suite gets compounded with the time the preceding suites took to execute), but that's another problem. I can change over to `microtime`, though I'm not sure if it matters here.

Should tests be made sequential?

dianashk replied:
True, this isn't a benchmarking tool, but a lot of requests take less than a second so you would see 0.

I think running them all in parallel is the way to go. It would be useful to see each request's round-trip time, as a way to gauge which types of requests might take longer than others. Agreed that total time doesn't mean very much here, so I'd be fine with losing that altogether.

sevko replied:
These are actually being recorded on a per-suite basis, not per-request, and 9 requests (the current size of `search.json`) end up taking something like 655ms, so milliseconds seems granular enough. Maybe let's ditch per-suite time and just get aggregate time for all suites? Per-suite times are misleading if we run in parallel, so I think we should lose them, but total time taken is still accurate.

dianashk commented:
* I like the ouput_generator abstraction. Feels like a stream might be appropriate here?
 * `test.js` would be better having just the command line args and config processing and then pull the test suite execution code into a separate file.
 * delete searchTest.js since it's not being used anymore.

sevko commented:
Addressed points 2 and 3. Not sure if a stream makes sense since we gather all of the test-results and statistics (in order to compute things like total number of passing/failing tests) before passing them to the output generator(s). The side-effect is that you don't see anything until all tests run to completion. Think it's worth implementing? Keep in mind that it wouldn't work for something like generating an email, so we'd have separate interfaces for the terminal and, for example, email output generators.

dianashk commented:
Agreed, it's not worth refactoring to streams for reasons listed.
 +1
